lib/src/trailer.rs

```diff
@@ -0,0 +1,338 @@
+// Copyright 2024 The Jujutsu Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Parsing trailers from commit messages.
+
+use itertools::Itertools as _;
+
+/// A key-value pair representing a trailer in a commit message, of the
+/// form `Key: Value`.
+#[derive(Debug, PartialEq, Eq, Clone)]
+pub struct Trailer {
+    /// trailer key
+    pub key: String,
+    /// trailer value
+    ///
+    /// It is trimmed at the start and the end but includes new line characters
+    /// (\n) and multi-line escape chars ( ) for multi line values.
+    pub value: String,
+}
+
+/// Parse the trailers from a commit message; these are simple key-value
+/// pairs, separated by a colon, describing extra information in a commit
+/// message; an example is the following:
+///
+/// ```text
+/// chore: update itertools to version 0.14.0
+///
+/// Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod
+/// tempor incididunt ut labore et dolore magna aliqua.
+///
+/// Co-authored-by: Alice <alice@example.com>
+/// Co-authored-by: Bob <bob@example.com>
+/// Reviewed-by: Charlie <charlie@example.com>
+/// Change-Id: I1234567890abcdef1234567890abcdef12345678
+/// ```
+///
+/// In this case, there are four trailers: two `Co-authored-by` lines, one
+/// `Reviewed-by` line, and one `Change-Id` line.
+pub fn parse_description_trailers(body: &str) -> Vec<Trailer> {
+    let (trailer, blank) = parse_trailers_and_blank(body);
```

Comment:
nit: use plural `trailers` (same on line 65 and line 77). sorry that i didn't think of this before